crates/biome_js_analyze/src/lint/nursery/no_substr.rs

```diff
@@ -0,0 +1,204 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+};
+use biome_console::markup;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsName, JsCallArguments, JsExpressionStatement, JsSyntaxToken, JsVariableStatement,
+};
+use biome_rowan::{declare_node_union, AstSeparatedList, BatchMutationExt, TextRange, TokenText};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Enforce the use of `slice()` over `substr()` and `substring()`.
+    ///
+    /// `slice()` is preferred over `substr()` and `substring()` because it is a more popular option with clearer behavior,
+    ///  and it has a consistent counterpart in arrays.
```

Comment:
```suggestion
    /// `String.slice()` is preferred over `String.substr()` and `String.substring()` because it is a more popular option with clearer behavior,
    ///  and it has a consistent counterpart in arrays.
```